pkgs/build-support/dart/pub2nix/package-graph.py

```diff
@@ -0,0 +1,58 @@
+"""
+https://github.com/dart-lang/pub/issues/4522
+This script generates a package_graph.json file.
+"""
+
+import json
+import os
+from pathlib import Path
+from urllib.parse import unquote, urlparse
+
+import yaml
+
+
+def get_package(pubspec_path: Path, dev_dependencies: bool = False):
+    with pubspec_path.open("r", encoding="utf-8") as f:
+        pubspec = yaml.safe_load(f)
```

Comment:
```suggestion
        pubspec = yaml.load(f, Loader=yaml.CSafeLoader)
```

Nixpkgs's pyyaml is built with libyaml, so we can use it for speed.